core/node/api_server/src/web3/namespaces/en.rs

```diff
@@ -191,4 +193,27 @@ impl EnNamespace {
             .read_whitelisted_tokens_for_aa_cache()
             .await)
     }
+
+    pub async fn get_protocol_version_impl(
+        &self,
+        version_id: Option<u16>,
+    ) -> Result<Option<ProtocolVersionInfo>, Web3Error> {
+        let mut storage = self.state.acquire_connection().await?;
+        let protocol_version_info = if let Some(id) = version_id {
+            storage
+                .protocol_versions_web3_dal()
+                .get_protocol_version_info_by_id(id)
+                .await
+                .map_err(DalError::generalize)?
+        } else {
+            Some(
+                storage
+                    .protocol_versions_web3_dal()
+                    .get_latest_protocol_version_info()
```

Comment:
Bikeshedding: Since `latest_semantic_version()` is used in the API server handlers, it may make sense to use it here as well (followed by `get_protocol_version_info_by_id`) instead of introducing a specialized method.